tests/unit/remote/test_remote_dir.py

```diff
@@ -72,20 +66,21 @@ def test_exists(remote):
         assert remote.exists(remote.path_info / path) == expected
 
 
+@pytest.mark.parametrize("remote", remotes, indirect=True)
 def test_walk_files(remote):
     files = [
         remote.path_info / "data/alice",
         remote.path_info / "data/alpha",
         remote.path_info / "data/subdir/1",
         remote.path_info / "data/subdir/2",
         remote.path_info / "data/subdir/3",
-        remote.path_info / "empty_file",
-        remote.path_info / "foo",
+        remote.path_info / "data/subdir/empty_file",
     ]
 
-    assert list(remote.walk_files(remote.path_info)) == files
+    assert list(remote.walk_files(remote.path_info / "data")) == files
 
 
+@pytest.mark.parametrize("remote", [S3Mocked], indirect=True)
 def test_copy_preserve_etag_across_buckets(remote):
```

Comment:
We don't need to create all those blobs for this test and `parametrize` looks a bit stretched. I suggest using `@mock_s3` and creating both remotes simply with `RemoteS3(...)` and not using `S3Mocked` here.

Reply:
Current approach is also fine, I guess.

Reply:
Keeping it this way for now, then. :)